src/libs/connectorLibs/getLauncherClient.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+
+import CozyClient, { useClient } from 'cozy-client'
+
+import { LauncherContext } from '/screens/home/HomeScreen'
+import { getErrorMessage } from '/libs/functions/getErrorMessage'
+
+/**
+ * Gets the client to be used by the launcher.
+ *
+ * The launcher is the webview that displays the login form for a given connector.
+ * This function creates a new client with the right token for the connector.
+ *
+ * @param {CozyClient} client - The main client
+ * @param {string} slug - The connector slug
+ * @param {React.Dispatch<React.SetStateAction<CozyClient | undefined>>} callback - A callback invoked with the new client
+ * @returns {CozyClient} - A new client
+ */
+const getLauncherClient = async (
+  client: CozyClient,
+  slug: string,
+  callback?: React.Dispatch<React.SetStateAction<CozyClient | undefined>>
+): Promise<CozyClient> => {
+  const { fetchKonnectorToken, uri } = client.getStackClient()
+
+  try {
+    const client = new CozyClient({
+      token: await fetchKonnectorToken(slug),
+      uri
+    })
+
+    callback?.(client)
+
+    return client
+  } catch (error) {
+    throw new Error(
+      `Failed to create launcher client for ${slug}.\n,
+      ${getErrorMessage(error)}`
+    )
+  }
+}
+
+export const useLauncherClient = (
```

Comment:
Should we put this entire file in the `hook` folder instead? The `connectorLibs` should be for connectors `contentScript` (if I understand it correctly)

Reply:
Agreed